src/main/java/com/denizenscript/depenizen/bukkit/commands/effectlib/EffectLibCommand.java

```diff
@@ -1,170 +1,167 @@
+    // Defaults to the linked player if unspecified.
+    // Some effects (such as the 'Line' effect) require an origin and a target, an effect's target can be a location or an entity.
+    // If an entity is specified, the effect will follow it.
+    //
+    // You can optionally use the 'for:' argument to show the effect to one player only.
+    //
+    // EffectLib effects can take additional data to change how the effect looks/works, you can specify that data in the form of a MapTag using the 'effect_data:' argument.
+    // See EffectLib's docs for further information on the available options for each effect: <@link url https://reference.elmakers.com/#effectlib>
+    //
+    // Note that this should only be used for displaying effects. for showing single particles, etc. use <@link command playeffect>.
     //
-    // Note that most users should instead use <@link command playeffect>.
+    // The effectlib command is ~waitable. Refer to <@link language ~waitable>.
+    // When ~waited for, the command will wait until the effect is done playing.
     //
     // @Tags
     // None
     //
     // @Usage
-    // Use to show a effect on the attached player in queue.
-    // - effectlib bleed duration:10s
+    // Use to show a helix of end rod sparks below the linked player for 5 seconds.
+    // - effectlib type:Helix effect_data:[duration=<duration[5s].in_milliseconds>;particle=end_rod;offset=0,-1.5,0]
     //
     // @Usage
-    // Use to show a effect on a target entity.
-    // - effectlib atom target:<player.target> duration:10s
+    // Use to show a line of electric sparks between 2 entities for 1 minute.
+    // - effectlib type:Line origin:<[originEntity]> target:<[targetEntity]> effect_data:[duration=<duration[1m].in_milliseconds>;particle=electric_spark]
     //
     // @Usage
-    // Use to show effect at a position of the player.
-    // - effectlib type:atom duration:10s location:<player.location>
+    // Use to show an atom effect at a location, and narrate a message after it's done playing.
+    // - ~effectlib type:Atom origin:<[location]>
+    // - narrate "It's done!"
     //
     // -->
 
-    private enum Action {
-        BLEED, ARC, ATOM
-    }
+    public static Warning durationArgument = new SlowWarning("effectlibCommandDuration", "The 'duration:<duration>' argument for the 'effectlib' command is deprecated in favour of the 'duration' key in the 'effect_data' map, refer to the meta docs for more information.");
+    public static Warning oldTargetArguments = new SlowWarning("effectlibCommandOldTarget", "The 'target:<entity>'/'location:<location>' arguments for the 'effectlib' command are deprecated in favour of the 'origin:<entity>/<location>' argument.");
+    public static Warning caseInsensitiveEffectName = new SlowWarning("effectlibCommandCaseInsensitiveType", "Case-Insensitive effect types in the 'effectlib' commands are deprecated. make sure you're using correct casing.");
```

Comment:
Feels weird for this to be a deprecation - should probably be the other way around if anything - aim to support case-insensitivity *better* rather than plan its removal.